com.musixmatch.Musixmatch.yaml

```diff
@@ -0,0 +1,46 @@
+app-id: com.musixmatch.Musixmatch
+base: org.electronjs.Electron2.BaseApp
+base-version: '18.08'
+runtime: org.freedesktop.Platform
+runtime-version: '18.08'
+sdk: org.freedesktop.Sdk
+command: musixmatch
+separate-locales: false
+rename-desktop-file: musixmatch.desktop
+rename-icon: musixmatch
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+modules:
+  - name: musixmatch-desktop
+    buildsystem: simple
+    build-options:
+      build-args:
+        - --share=network
+    build-commands:
+      - ar x musixmatch_*.deb
+      - rm -f musixmatch_*.deb
+      - tar xf data.tar.xz
+      - rm -f control.tar.gz data.tar.xz debian-binary
+      - cp -r usr/* opt/* /app
+      - chmod -R a-s,go+rX,go-w /app/Musixmatch
+      - rm -r /app/share/icons/hicolor/1024x1024
+      - desktop-file-edit --set-key=Exec --set-value='musixmatch %U' /app/share/applications/musixmatch.desktop
+      - install musixmatch.sh /app/bin/musixmatch
+      - install -Dm644 com.musixmatch.Musixmatch.appdata.xml /app/share/appdata/com.musixmatch.Musixmatch.appdata.xml
+    sources:
+      - type: shell
+        only-arches:
+          - x86_64
+        commands:
+          - curl -A "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0" -L -o musixmatch_3.3.1_amd64.deb https://adv.musixmatch.com/r/wwwmxm
```

Comment:
The EULA doesn't appear to let us distribute this binary directly. You should modify the manifest to use extra-data so that the end user downloads it directly.

> 1.1 Unless otherwise specifically provided by us, our grant of rights to you under these Terms is personal to you and gives you access to use the Application and the Services only on the Device on which the software was first installed. The licence is not transferrable to another person or another Device without our agreement in writing (which will only be given in exceptional circumstances) or if otherwise expressly provided in these Terms.

Reply:
Hi @nedrichards , I'm trying to switch to extra-data, using [Teamspeak repo](https://github.com/flathub/com.teamspeak.TeamSpeak) as reference, but when I run

`flatpak-builder --install build com.musixmatch.Musixmatch.yaml`

the application is not installed, I can't find com.musixmatch.Musixmatch folder in /var/lib/flatpak/app , despite the output does not show any error:

```
[... redacted ...]
FB: Using Flatpak version 1.2.4
Installing app/com.musixmatch.Musixmatch/x86_64/master
Pruning cache
```
I suspect that it fails downloading the application package; as you can see I use a custom curl command (with user-agent) to download it, because otherwise it downloads a file with this content:

`No platform specified and impossible to detect one`

Is there a way to set user-agent with extra-data source type? Or a workaround?

Reply:
Update:
I found an URL that allows curl to download the package without setting user-agent, but now installation fails because org.electronjs.Electron2.BaseApp does not have `ar` binary to extract deb when running apply_extra.

Is it recommended to use org.freedesktop.Sdk as runtime is this situation?

Reply:
I found a way to download a tar.gz instead of deb, so I can switch back to use org.freedesktop.Platform as runtime.